File: Python_OOP_SoftUni/Inheritance_lab/single_inheritance/dog.py
from animal import Animal


class Dog(Animal):
    def __init__(self, name):
        super().__init__(name)

    def bark(self):
        return f"The {self.name} is barking..."


if __name__ == "__main__":
    mammal = Animal("mammal")
    muttley = Dog("muttley")

    print(mammal.eat())
    print(muttley.bark())
    print(muttley.eat())

File: C_Sharp-OOP-SoftUni/PolymorphismEx/WildFarm/Models/Animals/Mammals/Dog.cs
using System;
using System.Collections.Generic;
using System.Text;
using WildFarm.Models.Foods;

namespace WildFarm.Models.Animals.Mammals
{
    public class Dog : Mammal
    {
        public Dog(string name, double weight,  string livingRegion) 
            : base(name, weight,  livingRegion)
        {
        }

        public override void AskForFood()
        {
            Console.WriteLine("Woof!");
        }


        public override void Eat(Food food)
        {
            if (food.GetType().Name != "Meat")
            {
                Console.WriteLine($"{this.GetType().Name} does not eat {food.GetType().Name}!");
            }
            else
            {
                this.Weight += 0.40 * food.Quantity;
                this.FoodEaten += food.Quantity;
            }
        }


        public override string ToString()
        {
            return $"{this.GetType().Name} [{this.Name}, {this.Weight}, {this.LivingRegion}, {this.FoodEaten}]";
        }
    }
}
